File: C/Algorithms/342.c
bool isPowerOfFour(int n) {
  if (((n & 1) || (n <= 0)) && (n != 1)) {
    return false;
  }
  float rs = log(n) / log(4);
  float diff = round(rs) - rs;
  return (diff < 0 ? -diff : diff) < 1e-10;
}

File: Javascript/Algorithms/342.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfFour = function(n) {
  if (((n & 1) || n <= 0) && (n != 1)) return false;

  const res = Math.log(n) / Math.log(4), diff = (Math.round(res) - res);
  return (diff < 0 ? -diff : diff) < 1e-10;
};